llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1439,6 +1496,103 @@ bool LoopVectorizationLegality::canVectorizeLoopNestCFG(
   return Result;
 }
 
+void LoopVectorizationLegality::recordExitingBlocks() {
+  SmallVector<BasicBlock *, 8> ExitingBlocks;
+  TheLoop->getExitingBlocks(ExitingBlocks);
+
+  SmallVector<BasicBlock *, 4> CountableExitingBBs;
+  PSE.getSE()->getCountableExitingBlocks(TheLoop, &CountableExitingBBs);
```

Comment:
I don't think we need to add this method to ScalarEvolution; I tried the following and all tests still passed:

```
  for (BasicBlock *BB1 : ExitingBlocks)
    if (isa<SCEVCouldNotCompute>(PSE.getSE()->getExitCount(TheLoop, BB1)))
      UncountableExitingBlocks.push_back(BB1);
    else
      CountableExitingBlocks.push_back(BB1);
```

If there is a difference in semantics (maybe about the predicates?) then could you please add a test to demonstrate.